packages/relay/src/lib/clients/cache/redisCache.ts

```diff
@@ -0,0 +1,164 @@
+ * limitations under the License.
+ *
+ */
+
+import { Logger } from 'pino';
+import { ICacheClient } from './ICacheClient';
+import { Registry } from 'prom-client';
+import { createClient, RedisClientType } from 'redis';
+import { RedisCacheError } from '../../errors/RedisCacheError';
+import constants from '../../constants';
+
+/**
+ * A class that provides caching functionality using Redis.
+ */
+export class RedisCache implements ICacheClient {
+  /**
+   * Configurable options used when initializing the cache.
+   *
+   * @private
+   */
+  private readonly options = {
+    // Max time to live in ms, for items before they are considered stale.
+    ttl: Number.parseInt(process.env.CACHE_TTL ?? constants.CACHE_TTL.ONE_HOUR.toString())
+  };
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private readonly logger: Logger;
+
+  /**
+   * The metrics register used for metrics tracking.
+   * @private
+   */
+  private readonly register: Registry;
+
+  private readonly client: RedisClientType;
+
+  /**
+   * Creates an instance of `RedisCache`.
+   *
+   * @param {Logger} logger - The logger instance.
+   * @param {Registry} register - The metrics registry.
+   */
+  public constructor(logger: Logger, register: Registry) {
+    this.logger = logger;
+    this.register = register;
+
+    const redisUrl = process.env.REDIS_URL!;
+    const reconnectDelay = parseInt(process.env.REDIS_RECONNECT_DELAY || '1000');
+
+    this.client = createClient({
+      url: redisUrl,
+      socket: {
+        reconnectStrategy: (retries: number) => {
+          const delay = retries * reconnectDelay;
+          logger.warn(`Trying to reconnect with Redis, ${retries} left. Delay is ${delay} ms...`);
+          return delay;
+        },
+      },
+    });
+    this.client.connect();
+
+    this.client.on('ready', function () {
+      logger.info(`Connected to Redis server (${redisUrl}) successfully!`);
+    });
+
+    this.client.on('error', function (error) {
+      const redisError = new RedisCacheError(error);
+      if (redisError.isSocketClosed()) {
+        logger.error(`Error occurred with Redis Connection when closing socket: ${redisError.message}`);
+      } else {
+        logger.error(`Error occurred with Redis Connection: ${redisError.fullError}`);
+      }
+    });
+  }
+
+  /**
+   * Retrieves a value from the cache.
+   *
+   * @param {string} key - The cache key.
+   * @param {string} callingMethod - The name of the calling method.
+   * @param {string} [requestIdPrefix] - The optional request ID prefix.
+   * @returns {Promise<any | null>} The cached value or null if not found.
+   */
+  async get(key: string, callingMethod: string, requestIdPrefix?: string | undefined) {
+    const result = await this.client.get(key);
+    if (result) {
+      this.logger.trace(
+        `${requestIdPrefix} returning cached value ${key}:${JSON.stringify(result)} on ${callingMethod} call`
+      );
+      //add metrics
```

Comment:
```suggestion
      //TODO: add metrics
```

so is easier to spot that something is missing here

Reply:
Done.